src/deploymentstatus.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_DEPLOYMENTSTATUS_H
+#define BITCOIN_DEPLOYMENTSTATUS_H
+
+#include <chain.h>
+#include <sync.h>
+#include <versionbits.h>
+
+#include <limits>
+
+extern RecursiveMutex cs_main;
```

Comment:
Any chance of just introducing a new lock for the versionbitscache, to avoid this semantically-cyclic dependency between validation and deploymentstatus?

Reply:
Yeah, can make `VersionBitsCache` be natively threadsafe, rather than having the caller manage a mutex (which should help if it ever ends up having some sort of shared lock too). I've done that, though I've added the lock as the first commit, since I think that makes the impact of the change clearer.